vizro-core/src/vizro/tables/dash_aggrid.py

```diff
@@ -0,0 +1,78 @@
+"""Module containing the standard implementation of `dash-ag-grid.AgGrid`."""
+
+import dash_ag_grid as dag
+import pandas as pd
+
+from vizro.models.types import capture
+from vizro.tables._utils import _set_defaults_nested
+
+FORMAT_CURRENCY_EU = """d3.formatLocale({
+  "decimal": ",",
+  "thousands": "\u00a0",
+  "grouping": [3],
+  "currency": ["", "\u00a0€"],
+  "percent": "\u202f%",
+  "nan": ""
+})"""
+
+DATA_TYPE_DEFINITIONS = {
+    "number": {
+        "baseDataType": "number",
+        "extendsDataType": "number",
+        "columnTypes": ["numberColumn", "rightAligned"],
+        "appendColumnTypes": True,
+        "valueFormatter": {"function": "params.value == null ? 'NaN' : String(params.value)"},
+    },
+    "dollar": {
+        "baseDataType": "number",
+        "extendsDataType": "number",
+        "valueFormatter": {"function": "d3.format('($,.2f')(params.value)"},
+    },
+    "euro": {
+        "baseDataType": "number",
+        "extendsDataType": "number",
+        "valueFormatter": {"function": f"{FORMAT_CURRENCY_EU}.format('$,.2f')(params.value)"},
+    },
+    "percent": {
+        "baseDataType": "number",
+        "extendsDataType": "number",
+        "valueFormatter": {"function": "d3.format(',.1%')(params.value)"},
+    },
+    "numeric": {
+        "baseDataType": "number",
+        "extendsDataType": "number",
+        "valueFormatter": {"function": "d3.format(',.1f')(params.value)"},
+    },
+}
+
+
+@capture("aggrid")
+def dash_ag_grid(data_frame=None, **kwargs):
+    """Implementation of `dash-ag-grid.AgGrid` with sensible defaults."""
+    defaults = {
+        "columnDefs": [{"field": col} for col in data_frame.columns],
+        "rowData": data_frame.apply(
+            lambda x: (
+                x.dt.strftime("%Y-%m-%d")  # set date columns to `dateString` for AGGrid filtering to function
+                if pd.api.types.is_datetime64_any_dtype(x)
+                else x
+            )
+        ).to_dict("records"),
+        "defaultColDef": {
+            # "editable": True, #do not set, as this may confuse some users
+            "resizable": True,
+            "sortable": True,
+            "filter": True,
+            "filterParams": {
+                "buttons": ["apply", "reset"],
+                "closeOnApply": True,
+            },
+        },
+        "dashGridOptions": {
+            "dataTypeDefinitions": DATA_TYPE_DEFINITIONS,
+            "animateRows": False,
```

Comment:
Why don't we want this? Animated rows sounds cool 😀

Reply:
Haha - that's my personal preference - I find the animation super annoying :face_with_spiral_eyes: 
You can give it a try -- the people at AG Grid think it makes for a "better user experience" :woman_shrugging:

Reply:
I was 50/50, it seems very snappy without, but somewhat fancy with it. Let's leave them out for now